app/services/proxy_status/base.rb

```diff
@@ -0,0 +1,45 @@
+module ProxyStatus
+  class Base
+    def initialize(proxy, opts = {}, api_class_name = 'Version')
+      @proxy = proxy
+      @cache_duration = opts[:cache_duration] || 3.minutes
+      @cache_status = opts[:cache_status].nil? ? true : opts[:cache_status]
+      @api_class_name = api_class_name
+    end
+
+    def revoke_cache!
+      # As memcached does not support delete_matched, we need to delete each
+      Rails.cache.delete(cache_key)
+    end
+
+    def cache_key
+      raise NotImplementedError, _("cache_key not implemented for ProxyStatus::#{self.class}")
```

Comment:
This is not translator friendly, but I guess it's fine since only developers should ever see this exception?